sonar-xml-plugin/src/main/java/org/sonar/plugins/xml/XmlSensor.java

```diff
@@ -97,19 +99,41 @@ private void runChecks(SensorContext context, XmlFile xmlFile, NewXmlFile newXml
       checks.all().stream()
         .filter(AbstractXmlCheck.class::isInstance)
         .map(AbstractXmlCheck.class::cast)
-        .forEach(check -> {
-          check.setRuleKey(checks.ruleKey(check));
-          check.validate(sourceCode);
-        });
-      saveIssue(context, sourceCode);
+        .forEach(check -> runCheck(check, sourceCode));
+      saveIssues(context, sourceCode);
     }
 
     checks.all().stream()
       .filter(NewXmlCheck.class::isInstance)
       .map(NewXmlCheck.class::cast)
-      .forEach(check -> check.scanFile(context, newXmlFile));
+      .forEach(check -> runCheck(context, check, newXmlFile));
+  }
+
+  private void runCheck(AbstractXmlCheck check, XmlSourceCode sourceCode) {
+    try {
+      check.setRuleKey(checks.ruleKey(check));
+      check.validate(sourceCode);
+    } catch (Exception e) {
+      logFailingRule(check.getRuleKey().rule(), sourceCode.getInputFile().uri(), e);
+    }
+  }
+
+  @VisibleForTesting
+  static void runCheck(SensorContext context, NewXmlCheck check, NewXmlFile newXmlFile) {
+    try {
+      check.scanFile(context, newXmlFile);
+    } catch (Exception e) {
+      logFailingRule(check.ruleKey(), newXmlFile.getInputFile().uri(), e);
+    }
+  }
+
+  private static void logFailingRule(String rule, URI fileLocation, Exception e) {
+    logError(String.format("Unable to execute rule %s on %s", rule, fileLocation), e);
+  }
 
-    saveSyntaxHighlighting(context, XMLHighlighting.highlight(newXmlFile), xmlFile.getInputFile());
+  private static void logError(String message, Exception e) {
+    LOG.warn(message);
```

Comment:
Should we append to message smth like `run with debug option to see more information`?

Reply:
I prefer to log using ERROR directly, and then provide the real stack then.